internal/ingress/controller/store/store_test.go

```diff
@@ -740,3 +740,105 @@ func TestUpdateSecretIngressMap(t *testing.T) {
 		}
 	})
 }
+
+func TestListIngresses(t *testing.T) {
+	s := newStore(t)
+
+	ing1 := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "testns",
+		},
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "demo",
+				ServicePort: intstr.FromInt(80),
+			},
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "foo.bar",
+				},
+			},
+		},
+	}
+	s.listers.Ingress.Add(ing1)
+
+	ingressToIgnore := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test2",
+			Namespace: "testns",
+			Annotations: map[string]string{
+				"kubernetes.io/ingress.class": "something",
+			},
+		},
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "demo",
+				ServicePort: intstr.FromInt(80),
+			},
+		},
+	}
+	s.listers.Ingress.Add(ingressToIgnore)
+
+	ingressWithoutPath := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test3",
+			Namespace: "testns",
+			Annotations: map[string]string{
+				"kubernetes.io/ingress.class": "nginx",
+			},
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "foo.bar",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "demo",
+										ServicePort: intstr.FromInt(80),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	s.listers.Ingress.Add(ingressWithoutPath)
+
+	ingressWithNginxClass := &v1beta1.Ingress{
```

Comment:
I think the names got mixed between this and previous ingress. Since the previous ingress is the one with `"kubernetes.io/ingress.class": "nginx",`